apps/sasquatch/src/projectDependency/java/com/microsoft/appcenter/sasquatch/activities/StorageActivity.java

```diff
@@ -173,6 +199,42 @@ private void updateStorageType(int position) {
         }
     }
 
+    public boolean saveArray(ArrayList<String> list, String name) {
+        SharedPreferences sp = getSharedPreferences(name, Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = sp.edit();
+        editor.putInt("Status_size", list.size());
+
+        for (int i = 0; i < list.size(); i++) {
+            editor.remove("Status_" + i);
+            editor.putString("Status_" + i, list.get(i));
+        }
+        return editor.commit();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        if (adapterUser != null) {
+            mUserDocumentList.clear();
+            mUserDocumentContents.clear();
+            String accountId = MainActivity.sSharedPreferences.getString(ACCOUNT_ID, null);
+            if (accountId != null) {
+                Storage.list(Constants.USER, Map.class).thenAccept(new AppCenterConsumer<PaginatedDocuments<Map>>() {
+                    @Override
+                    public void accept(PaginatedDocuments<Map> documents) {
+                        for (Document<Map> document : documents.getCurrentPage().getItems()) {
+                            mUserDocumentList.add(document.getId());
+                            mUserDocumentContents.add(Utils.getGson().toJson(document.getDocument()));
```

Comment:
Can getDocument return null?

Reply:
Yes on error, good point.
We need to do error management and maybe show a toast.